draft-ietf-taps-interface.md

```diff
@@ -1185,30 +1193,41 @@ number of Messages with a single call. This allows an application to provide
 backpressure to the transport stack when it is temporarily not ready to
 receive messages.
 
+Receive takes an optional minIncompleteLength argument. This value indicates
+the smallest partial Message data size that should be delivered. By default, this
+value is infinite, which means that only complete, atomic Messages will be delivered.
+If this value is set to some smaller value, the associated ReceiveHandler will
+be triggered only when at least that many bytes are available, the Message is
+complete, or the system needs to free up memory.
+
 Receive also takes an optional maxLength argument, the maximum size (in bytes
 of data) Message the application is currently prepared to receive. The default
 value for maxLength is infinite. If an incoming Message is larger than the
 minimum of this size and the maximum Message size on receive for
 the Connection's Protocol Stack, it will be received as a partial Message.
-Note that maxLength does not guarantee that the application will receive that
-many bytes if they are available; the interface may return partial Messages
-smaller than maxLength according to implementation constraints.
+Partial Messages are indicated by marking the endOfMessage boolean value
+as false in the ReceivedHandler. Note that maxLength does not guarantee that 
+the application will receive that many bytes if they are available; the interface may
+return partial Messages smaller than maxLength according to implementation constraints.
 
 ~~~
-Connection -> Received<Message>
+Connection -> Received<Message, messageData, endOfMessage>
```

Comment:
as with send, having a thing called a Message as well as messageData also on receive is... confusing. Renaming this to `MessageContext` would fix this at the expense of verbosity and slight obtuseness...

Reply:
I like MessageContext